Services/Statistics/SimulationStatistics.cs

```diff
@@ -110,6 +100,25 @@ public async Task CreateOrUpdateAsync(string simulationId, SimulationStatisticsM
             }
         }
 
+        public async Task UpdateAsync(string simulationId, SimulationStatisticsModel statistics)
+        {
+            var nodeId = this.clusterNodes.GetCurrentNodeId();
+            var statisticsRecordId = this.GetStatisticsRecordId(simulationId, nodeId);
+
+            var statisticsStorageRecord = this.GetStorageRecord(simulationId, statistics);
+
+            try
+            {
+                this.log.Debug("Updating statistics record", () => new { statisticsStorageRecord });
+                var record = await this.simulationStatisticsStorage.GetAsync(statisticsRecordId);
+                await this.simulationStatisticsStorage.UpsertAsync(statisticsStorageRecord, record.ETag);
+            }
+            catch (Exception e)
+            {
+                this.log.Error("Error on saving statistics records", e);
```

Comment:
do we need to log this into diagnostics logger as well?

Reply:
we dont need to log this.